### ISSUE TITLE
this project is fake as your dad coming back with the milk

### ISSUE THREAD

**Issue by radmanplays** [catfoolyou/Eaglercraft-1.12-Desktop-Runtime#9]
the CompileDesktopRuntime script just cleans the gradle build and it looks like you tried to port forge. 

POV being a skid 💀 :
![image](https://github.com/catfoolyou/Eaglercraft-1.12/assets/95340057/d6fdae50-86af-43c8-a30c-129c891a52e0)

**Comment by radmanplays** [comment]
and the server folder is not even the source code for the server, its a fcking compiled server.jar : 
![image](https://github.com/catfoolyou/Eaglercraft-1.12/assets/95340057/bf4d7944-db44-41d8-96a6-70b257a26df2)

**Comment by radmanplays** [comment]
@catfoolyou if you dont know java or how minecraft works please, just stop

**Comment by radmanplays** [comment]
why is there 2 assets folders? one in the main folder and one in the main/resources folder?

![image](https://github.com/catfoolyou/Eaglercraft-1.12/assets/95340057/e4d77e0c-30c6-4794-9372-b2f964b48686)

**Comment by catfoolyou** [comment]
My guy it works, just look in the releases. I can help you set this up if you want

First, forge is needed for shadersmod and Optifine to work. 
Second, the server folder doesn't have any source because its just shit for the server to work
Third, there are two assets folders because one is the actual assets and one is minecraft's "working directory"

`./gradlew clean build` cleans out existing compiled code and builds it again fyi.

**Comment by catfoolyou** [comment]
And how exactly am I skidding? This is just code from mcp 940, along with some of the 1.12 Optifine src added in.

**Comment by radmanplays** [comment]
then, you should rename it to "optifine src fork-1.12" bc its NOT possible to port forge and optifine and minecraft 1.12 and even if you manage to do that minecraft forge is NOT designed to be browser compatible and browser runtime cant read the java bytecode in .jar files

**Comment by catfoolyou** [comment]
Why? This project was never meant to be ported to the web. Its just the desktop runtime.

**Comment by catfoolyou** [comment]
I'm not planning to port it to the browser any time soon

**Comment by radmanplays** [comment]
"Eaglercraft" is minecraft ported to web, if you dont want to port it to web just change the repo's name.

**Comment by radmanplays** [comment]
the eaglercraft "desktop runtime" is minecraft but you can join websocket servers

**Comment by radmanplays** [comment]
@catfoolyou

**Comment by radmanplays** [comment]
> radman can you stop bothering the man it doesn't matter if he is wrong or not he is trying his best to make eaglercraft 1.12 and it doesn't matter if he doesn't port it to the web or not. AND also the word "eaglercraft" doesn't seem to even have a official meaning yet so why should you determine what it means?

i know, im just sick of kids pretending to "meak eaglercraft"

**Comment by catfoolyou** [comment]
Well I'm just trying to make a desktop runtime for now. It works and I have not claimed to have made something that I did not. I always thought that "Eaglercraft" meant "Portable Minecraft client with web and desktop versions".

> i know, im just sick of kids pretending to "meak eaglercraft"

I know, everyone is.

**Comment by catfoolyou** [comment]
> the eaglercraft "desktop runtime" is minecraft but you can join websocket servers

My client can connect to said servers without a websocket connection, and it works just fine.

**Comment by catfoolyou** [comment]
@radmanplays I still have not gotten a clear answer from aperson about whether or not I have been accepted onto the 1.12 development team, maybe if you could talk to him about it I could get on? I know Java and gradle, and I could help with rewriting the 1.12 source to work with teaVM.

**Comment by lax1dude** [comment]
There is no aperson 1.12 source, he thought he could feed MCP 1.12 to my 1.8 buildtools and apply my 1.8 patch files and be 80% done, promoted his repo and discord server before actually trying to do that, then when he tried to do that it didn't work and he took it all private. If you look at the commit history channel in his discord server he's started over twice and none of these iterations have ever been compiled. If I had to guess he's spent maybe 50 hours at the most actually focused on 1.12 over the past year and otherwise he doesn't even work on it. Meanwhile I spent 6 months working 6 to 12 hours a day before I even announced EaglercraftX 1.8 and that was two months before I even released u0. Its a long process that takes a lot of attention to detail and that's clearly not the mindset he approaches his with.

**Comment by catfoolyou** [comment]
That's a lot of effort put into that project. 

I always thought the 1.12 version would be done by replacing all of the 1.8 MC source in the 1.8 workspace, then modifying it to work with the rest of Eagler's code like the 1.8 src was modified. Do you think that would work?

**Comment by orphanlol** [comment]
> There is no aperson 1.12 source, he thought he could feed MCP 1.12 to my 1.8 buildtools and apply my 1.8 patch files and be 80% done, promoted his repo and discord server before actually trying to do that, then when he tried to do that it didn't work and he took it all private. If you look at the commit history channel in his discord server he's started over twice and none of these iterations have ever been compiled. If I had to guess he's spent maybe 50 hours at the most actually focused on 1.12 over the past year and otherwise he doesn't even work on it. Meanwhile I spent 6 months working 6 to 12 hours a day before I even announced EaglercraftX 1.8 and that was two months before I even released u0. Its a long process that takes a lot of attention to detail and that's clearly not the mindset he approaches his with.

lax1dude has come back!! (12 year olds complaining about eagler is wild)

note to skids and 12yr olds reading: lax doesn't owe you shit !! entitled little shits!!! (this is why he doesn't say how to port versions and rewrite shit)


tl;dr of lax's respone: aperson is a skid who is stupid af and thought giving 1.12 mcp to 1.8 buildtools would work and promoted shit that doesn't even work, it takes a long time to even port a minecraft version, stop clout chasing, and its a long process that requires lots of attention to detail ~~ that's clearly not the mindset he approaches his with.~~ , but his approach is like clout-chasing

**Comment by Vexb0t** [comment]
> > There is no aperson 1.12 source, he thought he could feed MCP 1.12 to my 1.8 buildtools and apply my 1.8 patch files and be 80% done, promoted his repo and discord server before actually trying to do that, then when he tried to do that it didn't work and he took it all private. If you look at the commit history channel in his discord server he's started over twice and none of these iterations have ever been compiled. If I had to guess he's spent maybe 50 hours at the most actually focused on 1.12 over the past year and otherwise he doesn't even work on it. Meanwhile I spent 6 months working 6 to 12 hours a day before I even announced EaglercraftX 1.8 and that was two months before I even released u0. Its a long process that takes a lot of attention to detail and that's clearly not the mindset he approaches his with.
> 
> lax1dude has come back!! (12 year olds complaining about eagler is wild)
> 
> note to skids and 12yr olds reading: lax doesn't owe you shit !! entitled little shits!!! (this is why he doesn't say how to port versions and rewrite shit)
> 
> tl;dr of lax's respone: aperson is a skid who is stupid af and thought giving 1.12 mcp to 1.8 buildtools would work and promoted shit that doesn't even work, it takes a long time to even port a minecraft version, stop clout chasing, and its a long process that requires lots of attention to detail ~that's clearly not the mindset he approaches his with.~ , but his approach is like clout-chasing

hi

**Comment by lax1dude** [comment]
This is the only other comment I’m gonna make about this repo but can you maybe come up with an original name for it instead of using my brands to promote it?

I don’t know how you decided the definition of “eaglercraft desktop runtime” is just “free vanilla minecraft with websockets” but that is not what that term means at all. If you look at old 1.5.2 commits it was originally called “Eaglercraft Debug Runtime” and the purpose was to be a desktop-based staging environment for debugging some of the browser client’s most important features more easily than they can be when the client is fully compiled to javascript, the purpose of it was not to have a desktop minecraft client compatible with eaglercraft servers.

You already said you probably won’t actually port this to browser in the end, and if you did try to port this through copy/pasting a ton of my code it won't work because of forge and optifine and shaders mod, therefore I don’t see how this can be compared to the real “eaglercraft desktop runtime” if it will never be turned into a working browser client.

**Comment by orphanlol** [comment]
lax1dude's comment is missing, so I'll quote it:
> This is the only other comment I’m gonna make about this repo but can you maybe come up with an original name for it instead of using my brands to promote it?
>
> I don’t know how you decided the definition of “eaglercraft desktop runtime” is just “free vanilla minecraft with websockets” but that is not what that term means at all. If you look at old 1.5.2 commits it was originally called “Eaglercraft Debug Runtime” and the purpose was to be a desktop-based staging environment for debugging some of the browser client’s most important features more easily than they can be when the client is fully compiled to javascript, the purpose of it was not to have a desktop minecraft client compatible with eaglercraft servers.

**Comment by catfoolyou** [comment]
For the record (and this is among the 15 or so guys who play Eaglercraft at my school), the web port is used exclusively by people who have chromebooks (which is most of us) and the "desktop runtime" is used by me and another guy (he was the one who helped me set most of this thing up) who take CS classes because the web version doesn't give nearly as much FPS on those shitty old PCs.

**Comment by catfoolyou** [comment]
I never knew that the desktop runtime was originally for debug only purposes, though I had speculations about that because I had run across the name "Eaglercraft debug runtime" multiple times.

**Comment by catfoolyou** [comment]
If Lax1dude wishes so, I most certainly will rename this repo and possibly even take it down. This whole 1.12 thing started as a hobby project amongst me and my friends to upgrade our clients and servers to 1.12. It was never meant to get this big, the only reason I put it on Github in the first place is that it is easier for people to download the source and work with it.

Again, I am willing to do whatever Lax wishes with this repo, it was never meant to turn into this.

**Comment by py660** [comment]
dang what a flame war

also hi

**Comment by TommyWooo** [comment]
Wait... I'm a little dumb, but how would you play this? Ik theres no offlineDownload, but do I run this all in something like, let's say... Repl?

**Comment by py660** [comment]
So I have yet to compile this program, but I would assume that you need an actual computer (not a repl or a chromebook) and command line access.

**Comment by catfoolyou** [comment]
You could theoretically run this off a chromebook if you have java installed and are running linux. The target platforms of this thing are Windows and Linux PCs, there is no web port yet (and will probably never be).

**Comment by DevidIsCool** [comment]
How do i compile the teavm verison of eaglercraft?

**Comment by TommyWooo** [comment]
Self explanatory. Look at the readme and run the 'gradlew' stuff if you're on Linux. If you don't know how to do this I can try setting up a space or something like that and share it with you.

**Comment by TommyWooo** [comment]
I would recommend using Gitpod to run this since if you mess up compiling it somehow you don't have to dump your whole computer. Also use Gitpod if you use a school computer.

**Comment by catfoolyou** [comment]
> How do i compile the teavm verison of eaglercraft?

Dude. The teavm version is still indev and doesnt compile. Come back in a couple of months.

**Comment by DevidIsCool** [comment]
kk
